go.mod

```diff
@@ -1,8 +1,8 @@
 module sigs.k8s.io/cluster-api-provider-azure
 
-go 1.22.7
+go 1.23.0
```

Comment:
Can this go bump be its own PR that we merge before this one?

Reply:
This comment is OBE based on - https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5421#issuecomment-2688701473